app/helpers/browse_helper.rb

```diff
@@ -1,15 +1,43 @@
 module BrowseHelper
+  def element_icon(type, object)
+    icon_data = { :filename => "#{type}.svg" }
+
+    unless object.redacted?
+      target_tags = object.tags.find_all { |k, _v| BROWSE_ICONS.key? k.to_sym }.sort
+      title = target_tags.map { |k, v| "#{k}=#{v}" }.to_sentence unless target_tags.empty?
+
+      target_tags.each do |k, v|
+        k = k.to_sym
+        v = v.to_sym
+        if v != :* && BROWSE_ICONS[k].key?(v)
+          icon_data = BROWSE_ICONS[k][v]
+        elsif BROWSE_ICONS[k].key?(:*)
```

Comment:
I'd suggest not allowing a wildcard key only match to override a more specific key=value match here:

```suggestion
        elsif icon_data.nil? && BROWSE_ICONS[k].key?(:*)
```

Reply:
Now I have default priorities of 1 for type icons, 10 for `k=*` and 100 for `k=v`; and it's possible to set a different priority in `browse_icons.yml`. I did it to control which icon is shown in case of `building=yes` + `shop=unknown_shop_type`.